### PR TITLE
Use dummy subscriptions for fdroid flavor

### PR DESCRIPTION
**Pull request by lmac012**
Task/Issue URL: https://app.asana.com/0/1205648422731273/1207600736669442/f

### Description

This PR adds dummy implementation of subscriptions-api and makes the fdroid flavor of the app depend on it instead of the actual implementation. The goal is to remove dependency on Google Play Billing Library from the F-Droid builds of the app.

### Steps to test this PR

See task.

### CODE REVIEW

**Comment by lmac012** [comment]
* **#4683** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4683?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lmac012 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: subscriptions/subscriptions-dummy-impl/build.gradle [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    coreLibraryDesugaring Android.tools.desugarJdkLibs
+}
+
+android {
+    namespace "com.duckduckgo.subscriptions.dummy.impl"
+    anvil {
+        generateDaggerFactories = true // default is false
+    }
+    lint {
```
Comment (marcosholgado) [inline subscriptions/subscriptions-dummy-impl/build.gradle:42]: The baseline can be removed

File: subscriptions/subscriptions-dummy-impl/lint-baseline.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (marcosholgado) [inline subscriptions/subscriptions-dummy-impl/lint-baseline.xml:1]: And the xml file too :)

**Review by anikiki — APPROVED** [review]
LGTM :+1:

**Pull request merged**